File: dailyQuestion/2020-06/06-23/golang/solution_bit.go
package main

import (
	"fmt"
	"math/big"
)

func addBinary(a string, b string) string {
	ai, _ := new(big.Int).SetString(a, 2)
	bi, _ := new(big.Int).SetString(b, 2)
	ai.Add(ai, bi)
	return ai.Text(2)
}

func main() {
	a := "11"
	b := "1"
	result := addBinary(a, b)
	fmt.Println(result)
}

File: dailyQuestion/2020-06/06-23/python/solution_bit.py
def addBinary(a: str, b: str) -> str:
    a = int(a, 2)
    b = int(b, 2)
    while b:
        result = a ^ b
        carry = (a & b) << 1
        a = result
        b = carry
    return bin(a)[2:]

if __name__ == "__main__" :
    a = "11"
    b = "1"
    c = "11".rjust(2,"0")
    result = addBinary(a,b)
    print(result)